src/Block/BlockRenderer.php

```diff
@@ -79,6 +82,10 @@ public function render(BlockContextInterface $blockContext, ?Response $response
                 ), compact('exception'));
             }
 
+            if ($exception instanceof RuntimeError && !empty($exception->getPrevious()) && $exception->getPrevious() instanceof BlockNotFoundException) {
```

Comment:
I don't think this code belongs here. We have ExceptionStrategyManager for that I guess.

Reply:
Yes this could be here instead https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Exception/Strategy/StrategyManager.php#L107-L125